gadgets/trace_open/test/trace_open_test.go

```diff
@@ -46,63 +48,97 @@ type traceOpenEvent struct {
 
 func TestTraceOpen(t *testing.T) {
 	gadgettesting.RequireEnvironmentVariables(t)
+	InitTest(t)
 
-	runtime := "docker"
-	containerFactory, err := containers.NewContainerFactory(runtime)
+	containerFactory, err := containers.NewContainerFactory(Runtime)
 	require.NoError(t, err, "new container factory")
-
 	containerName := "test-trace-open"
-	containerImage := "docker.io/library/busybox"
-
-	testContainer := containerFactory.NewContainer(
-		containerName,
-		"while true; do setuidgid 1000:1111 cat /dev/null; sleep 0.1; done",
-		containers.WithContainerImage(containerImage),
-	)
-
-	testContainer.Start(t)
-	t.Cleanup(func() {
-		testContainer.Stop(t)
-	})
-
-	traceOpenCmd := igrunner.New(
-		"trace_open",
-		igrunner.WithFlags(fmt.Sprintf("--runtimes=%s", runtime), "--timeout=5"),
-		igrunner.WithValidateOutput(
-			func(t *testing.T, output string) {
-				expectedEntry := &traceOpenEvent{
-					CommonData: eventtypes.CommonData{
-						Runtime: eventtypes.BasicRuntimeMetadata{
-							RuntimeName:        eventtypes.String2RuntimeName(runtime),
-							ContainerName:      containerName,
-							ContainerID:        testContainer.ID(),
-							ContainerImageName: containerImage,
-						},
-					},
-					Comm:  "cat",
-					Fd:    3,
-					Err:   0,
-					FName: "/dev/null",
-					Uid:   1000,
-					Gid:   1111,
-					Flags: 0,
-					Mode:  0,
+	containerImage := "docker.io/library/busybox:latest"
+	containerCommand := "while true; do setuidgid 1000:1111 cat /dev/null; sleep 0.1; done"
+
+	var expectedCommonData eventtypes.CommonData
+	var ns string
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+
+	switch DefaultTestComponent {
+	default:
+		t.Fatalf("invalid runtime: %s. Valid values should contain: ig, kubectl-gadget", DefaultTestComponent)
```

Comment:
I think we can move this test to InitTest so each gadget doesn't have to worry about this.